File: student-management-system/src/main/java/com/student/studentmanagementsystem/service/CourseService.java
package com.student.studentmanagementsystem.service;

import java.util.List;

import com.student.studentmanagementsystem.entities.Course;

public interface CourseService {
    List<Course> getAllCourses();
    Course getCourseById(Long course_id);
    Course saveCourse(Course course);
    Course updateCourse(Long course_id, Course updatedCourse);
    void deleteCourse(Long course_id);
}

File: student-management-system-frontend/src/app/services/course.service.ts
// course.service.ts

import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Course } from '../model/course.model';
import { environment } from '../../environments/environment';

@Injectable({
  providedIn: 'root',
})
export class CourseService {
  // private apiUrl = 'http://localhost:8080/api/courses';
  private apiUrl = `${environment.apiUrl}/courses`;


  constructor(private http: HttpClient) {}

  getAllCourses(): Observable<Course[]> {
    return this.http.get<Course[]>(this.apiUrl);
  }

  getCourseById(id: number): Observable<Course> {
    return this.http.get<Course>(`${this.apiUrl}/${id}`);
  }
  
  addCourse(course: Course): Observable<Course> {
    return this.http.post<Course>(this.apiUrl, course);
  }

  updateCourse(id: number, course: Course): Observable<Course> {
    return this.http.put<Course>(`${this.apiUrl}/${id}`, course);
  }

  deleteCourse(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
